<!DOCTYPE html>
<html>
  <head>
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>munichimputation.readthedocs</title>
  </head>
  <body>
     <h1>Munich Imputation Server</h1>
     <a href="http://146.107.169.108:8082">Munich Imputation Server</a> provides a free genotype imputation service using <a href="https://genome.sph.umich.edu/wiki/Minimac4">Minimac4</a>. You can upload phased or unphased GWAS genotypes and receive phased and imputed genomes in return. Our server offers imputation from 1000 Genomes (Phase 1 and 3), CAAPA, <a href="http://www.haplotype-reference-consortium.org/">HRC</a> and the <a href="https://topmed.nhlbi.nih.gov/">TOPMed</a> reference panel. For all uploaded datasets an extensive QC is performed.     
     <br><br>
     
     <img src="munich_imputation_server.png" class="img-rounded" width="900px">>
   
     <p>Please cite this paper if you use Munich Imputation Server in your publication:</p>  
     
     <ul>
        <li>DaS <S></S>, Forer L, Schönherr S, Sidore C, Locke AE, Kwong A, Vrieze S, Chew EY, Levy S, McGue M, Schlessinger D, Stambolian D, Loh PR, Iacono WG, Swaroop A, Scott LJ, Cucca F, Kronenberg F, Boehnke M, Abecasis GR, Fuchsberger C. <a href="https://pubmed.ncbi.nlm.nih.gov/27571263/">Next-generation genotype imputation service and methods.</a> Nature Genetics 48, 1284–1287 (2016).</li>
       </ul>
 
      <h2>Getting Started</h2>
      To use Munich Imputation Server, a <a href="http://146.107.169.108:8082/index.html#!pages/register">registration</a> is required. We send an activation mail to the provided address. Please follow the instructions in the email to activate your account. If it doesn't arrive, ensure you have entered the correct email address and check your spam folder.
      <br>
      <p><b>After the email address has been verified, the service can be used without any costs.</b></p>
      
      <p>Please cite this paper if you use Munich Imputation Server in your GWAS study:</p>
            <ul>
               <li>Das S, Forer L, Schönherr S, Sidore C, Locke AE, Kwong A, Vrieze S, Chew EY, Levy S, McGue M, Schlessinger D, Stambolian D, Loh PR, Iacono WG, Swaroop A, Scott LJ, Cucca F, Kronenberg F, Boehnke M, Abecasis GR, Fuchsberger C. <a href="https://pubmed.ncbi.nlm.nih.gov/27571263/">Next-generation genotype imputation service and methods.</a> Nature Genetics 48, 1284–1287 (2016).</li>
             </ul>  
      
      
      <h2>Setup your first imputation job</h2>
      <p>Please <a href="http://146.107.169.108:8082/index.html#!pages/login">login</a> with your credentials and click on the <b>Run</b> tab to start a new imputation job. The submission dialog allows you to specify the properties of your imputation job.</p>
      
      <img src="run_job_screenshot.png" class="img-rounded" width="900px">
      
      <p>The following options are available:</p>

      <h3>Reference Panel</h3>
      <p>Our server offers genotype imputation from different reference panels. The most accurate and largest panel is <b>HRC (Version r1.1 2016)</b>. Please select one that fulfills your needs and supports the population of your input data:
         
      <ul>
         <li>HRC (Version r1.1 2016)</li>
         <li>HLA Imputation Panel: two-field (four-digit) and G-group resolution</li>
         <li>HRC (Version r1 2015)</li>
         <li>1000 Genomes Phase 3 (Version 5)</li>
         <li>HapMap 2</li>
      </ul>
         
         <p>More details about all available reference panels can be found here.</p>

         <h3>Input files</h3>
         <p>VCF files can be uploaded either from your local disk or by specifying a remote sftp / HTTPS location. In both cases, several files or several locations can be selected at once. Please make sure that all files fulfill the requirements.</p>
         <h3>Upload VCF files from your computer</h3>
         <p>When using the file upload, data is uploaded from your local file system to Michigan Imputation Server. By clicking on <b>Select Files</b> an open dialog appears where you can select your VCF files:</p>
         <h3>Import VCF files via SFTP</h3>
         <p>An other convenient way to use your data is by importing it directly from a remote SSH server. This can be achieved by selecting <b>Secure File Transfer Protocol (SFTP)</b>.<br><br>

         After clicking on the <b>Import Files</b> button, a new dialog appears where you can enter your SFTP credentials and the URLs of your files or folders. A URL consists of the server address followed by the full Unix path. A path can be either point to a folder or to a file. Several paths can be specified in consecutive lines.</p>
         
         <img src="sftp_screenshot.png" class="img-rounded" width="900px">

         <h3>Import VCF files via HTTPS</h3>
         <p>Data can also be imported by using HTTP(S) link. This can be achieved by selecting <b>URLs (HTTP)</b>.</p>
         <p>After clicking on the Add URLs button, a new dialog appears where you can add the links.</p>

         <h3>Build</h3>
         <p>Please select the build of your data. Currently the options <b>hg19</b> and <b>hg38</b> are supported. Michigan Imputation Server automatically updates the genome positions (liftOver) of your data. All reference panels except TOPMed are based on hg19 coordinates.         </p>
      
          <h3>rsq Filter</h3>
          <p>To minimize the file size, Michigan Imputation Server includes a r2 filter option, excluding all imputed SNPs with a r2-value (= imputation quality) smaller then the specified value.</p>
  
         <h3>Phasing</h3>
         <p>If your uploaded data is <i>unphased,</i> Eagle v2.4 will be used for phasing. In case your uploaded VCF file already contains phased genotypes, please select the "No phasing" option.</p>
         
         <table border="1" cellpadding="1" cellspacing="0">
            <tr>
               <th>Algorithm</th>
               <th>Description</th>
            </tr>
            <tr>
               <td>Eagle v2.4</td>
               <td>The <a href="https://alkesgroup.broadinstitute.org/Eagle/">Eagle</a> algorithm estimates haplotype phase using the HRC reference panel. This method is also suitable for single sample imputation. After phasing or imputation you will receive phased genotypes in your VCF files.</td>
            </tr>
         </table>
         
         <h3>Population</h3>
         <p>Please select the population of your uploaded samples. This information is used to compare the allele frequencies between your data and the reference panel. Please note that not every reference panel supports all sub-populations.</p>

         <table border="1" cellpadding="1" cellspacing="0">
            <tr>
               <th>Population</th>
               <th>Supported Reference Panels</th>
            </tr>
            <tr>
               <td>AFR</td>
               <td>all</td>
            </tr>
            <tr>
               <td>AMR</td>
               <td>all</td>
            </tr>
            <tr>
               <td>EUR</td>
               <td>all</td>
            </tr>
            <tr>
               <td>Mixed</td>
               <td>all</td>
            </tr>
            <tr>
               <td>AA</td>
               <td>CAAPA</td>
            </tr>
            <tr>
               <td>ASN</td>
               <td>1000 Genomes Phase 1 (Version3)</td>
            </tr>
            <tr>
               <td>EAS</td>
               <td>1000 Genomes Phase 3 (Version5)</td>
            </tr>
            <tr>
               <td>SAS</td>
               <td>1000 Genomes Phase 3 (Version5)</td>
            </tr>
         </table>
         <p>In case your population is not listed or your samples are from different populations, please select <b>Mixed</b> to skip the allele frequency check. For mixed populations, no QC-Report will be created.</p>
         
         <h3>Mode</h3>
         <p>Please select if you want to run <b>Quality Control & Imputation, Quality Control & Phasing Only</b> or <b> Control Only</b>.</p>
      
         <h3>AES 256 encryption</h3>
         <p>All Imputation Server results are encrypted by default. Please tick this checkbox if you want to use AES 256 encryption instead of the default encryption method. Please note that AES encryption does not work with standard unzip programs. We recommend to use 7z instead.</p>
      
         <h2>Start your imputation job</h2>
         <p>After confirming our Terms of Service, the imputation process can be started immediately by clicking on <b>Start Imputation</b>. Input Validation and Quality Control are executed immediately to give you feedback about the data-format and its quality. If your data passed this steps, your job is added to our imputation queue and will be processed as soon as possible. You can check the position in the queue on the job summary page.</p>
      
         <p>We notify you by email as soon as the job is finished or your data don't pass the Quality Control steps.</p>
      
         <h3>Input Validation</h3>
         <p>In a first step we check if your uploaded files are valid and we calculate some basic statistics such as amount of samples, chromosomes and SNPs.</p>
       
         <p>After Input Validation has finished, basic statistics can be viewed directly in the web interface.</p>
      
         <p>If you encounter problems with your data please read this tutorial about Data Preparation to ensure your data is in the correct format.</p>
  
         <h3>Quality Control</h3>
         <p>In this step we check each variant and exclude it in case of:</p>

         <ol>
            <li>contains invalid alleles</li>
            <li>duplicates</li>
            <li>indels</li>
            <li>monomorphic sites</li>
            <li>allele mismatch between reference panel and uploaded data</li>
            <li>SNP call rate < 90%</li>
         </ol>

         <p>All filtered variants are listed in a file called statistics.txt which can be downloaded by clicking on the provided link. More informations about our QC pipeline can be found here.</p>

         <p>If you selected a population, we compare the allele frequencies of the uploaded data with those from the reference panel. The result of this check is available in the QC report and can be downloaded by clicking on qcreport.html.</p>

         <h3>Pre-phasing and Imputation</h3>
         <p>Imputation is achieved with Minimac4. The progress of all uploaded chromosomes is updated in real time and visualized with different colors.</p>
         
         <img src="Pre-phasing_imputation.png" class="img-rounded" width="900px">>

         <h3>Data Compression and Encryption</h3>
         <p>If imputation was successful, we compress and encrypt your data and send you a random password via mail.</p>

         <img src="data_compression_encryption.png" class="img-rounded" width="900px">

         <p>This password is not stored on our server at any time. Therefore, if you lost the password, there is no way to resend it to you.</p>

         <h3>Download results</h3>
         <p>The user is notified by email, as soon as the imputation job has finished. A zip archive including the results can be downloaded directly from the server. To decrypt the results, a one-time password is generated by the server and included in the email. The QC report and filter statistics can be displayed and downloaded as well.</p>

         <h3>Download via a web browser</h3>
         <p>All results can be downloaded directly via your browser by clicking on the filename.</p>

         <img src="download_web_browser.png" class="img-rounded" width="900px">

         <p>In order to download results via the commandline using wget or aria2 you need to click on the share symbol (located right to the file size) to get the needed private links.</p>

         <img src="share-data.png" class="img-rounded" width="500px">

         <p>A new dialog appears which provides you the private link. Click on the tab wget command to get a copy & paste ready command that can be used on Linux or MacOS to download the file in you terminal:</p>

         <h3>Download all results at once</h3>
         <p>To download all files of a folder (for example folder Imputation Results) you can click on the share symbol of the folder:</p>

         <img src="download_all_results.png" class="img-rounded" width="900px">

      </body>
</html>



