<!DOCTYPE html>
<html>
<style>
    p.ridge {
        border-style: ridge;
        border-width: 1px;
        background-color: lightgrey;
        width: 650px;
        }
</style>

<h2>Data preparation</h2>

Michigan Imputation Server accepts VCF files compressed with <a href="http://www.htslib.org/doc/tabix.html" style="text-decoration: none">bgzip</a>. Please make sure the following requirements are met:

<ul>
    <li>Create a separate vcf.gz file for each chromosome.</li>
    <li>Variations must be sorted by genomic position.</li>
    <li>GRCh37 or GRCh38 coordinates are required.</li>
</ul>

<h3>Quality Control for HRC, 1000G and CAAPA imputation</h3>
Will Rayner provides a great toolbox to prepare data: <a href="https://www.well.ox.ac.uk/~wrayner/tools/" style="text-decoration: none">HRC or 1000G Pre-imputation Checks.</a>
<br><br>
The main steps for HRC are:

<h3>Download tool and sites</h3>
<p class="ridge">wget http://www.well.ox.ac.uk/~wrayner/tools/HRC-1000G-check-bim-v4.2.7.zip<br>
wget ftp://ngs.sanger.ac.uk/production/hrc/HRC.r1-1/HRC.r1-1.GRCh37.wgs.mac5.sites.tab.gz</p>

<h3>Convert ped/map to bed</h3>
<p class="ridge">plink --file (input-file) --make-bed --out (output-file)(/output-file)</p>

<h3>Create a frequency file</h3>
<p class="ridge">plink --freq --bfile (input) --out (freq-file)</p>


<h3>Execute script</h3>
<p class="ridge">perl HRC-1000G-check-bim.pl -b (bim file) -f (freq-file) -r HRC.r1-1.GRCh37.wgs.mac5.sites.tab -h
sh Run-plink.sh</p>

<h3>Create vcf using <a href="https://genome.sph.umich.edu/wiki/VcfCooker" style="text-decoration: none">VcfCooker</a></h3>
<p class="ridge">vcfCooker --in-bfile (bim file) --ref (reference.fasta)  --out (output-vcf) --write-vcf
bgzip (output-vcf)</p>

<h2>Additional Tools</h2>

<h3>Convert ped/map files to VCF files</h3>

Several tools are available: <a href="https://www.cog-genomics.org/plink2/" style="text-decoration: none">plink2,</a> <a href="https://samtools.github.io/bcftools/" style="text-decoration: none">BCFtools</a> or <a href="https://genome.sph.umich.edu/wiki/VcfCooker" style="text-decoration: none">VcfCooker.</a>

<p class="ridge">plink --ped study_chr1.ped --map study_chr1.map --recode vcf --out study_chr1</p>

Create a sorted vcf.gz file using <a href="https://samtools.github.io/bcftools/" style="text-decoration: none">BCFtools:</a><br> <br>

<p class="ridge">bcftools sort study_chr1.vcf -Oz -o study_chr1.vcf.gz</p>

<h3>CheckVCF</h3>

Use <a href="https://github.com/zhanxw/checkVCF" style="text-decoration: none">checkVCF</a> to ensure that the VCF files are valid. checkVCF proposes "Action Items" (e.g. upload to sftp server), which can be ignored. Only the validity should be checked with this command.<br> <br>

<p class="ridge">checkVCF.py -r human_g1k_v37.fasta -o out mystudy_chr1.vcf.gz</p>


