<html>
    <style>
        table {
          border-collapse: collapse;
          width: 40%;
        }
        
        th, td {
          text-align: left;
          padding: 8px;
        }
        
        tr:nth-child(even) {
          background-color: #D6EEEE;
        }
    </style>

<h2>Four-digit Multi-ethnic HLA v1 (2021)</h2>
The multi-ethnic HLA panel consists of 36,586 haplotypes of five global populations. We have released the panel at both two-field (four-digit) and G-group resolution.<br> <br>

<table border="1" cellpadding="5" cellspacing="0">
    <tr>
        <th></th>
        <th></th>
    </tr>
    <tr>
        <td>Number of Samples</td>
        <td>18,293</td>
    </tr>
    <tr>
        <td>Sites (chr6)</td>
        <td>56,310 (1,781 HLA alleles; 4,513 HLA amino acids; 10,924 SNPs within HLA; 39,092 scaffold SNPs)</td>
    </tr>
    <tr>
        <td>Chromosomes</td>
        <td>6:28,000,000-34,000,000</td>
    </tr>
    <tr>
        <td>Website</td>
        <td>
            <a href="https://github.com/immunogenomics/HLA-TAPAS" style="text-decoration: none">https://github.com/immunogenomics/HLA-TAPAS</a>
        </td>
    </tr>
</table>
<br>

Please cite this manuscript if you would like to include imputed results from the panel in your work:<br> <br>

Luo, Y., Kanai, M., Choi, W., Li, X., Yamamoto, K., Ogawa, K., Gutierrez-Arcelus, M., Gregersen, P. K., Stuart, P. E., Elder, J. T., Fellay, J., Carrington, M., Haas, D. W., Guo, X., Palmer, N. D., Chen, Y.-D. I., Rotter, J. I., Taylor, K. D., Rich, S., â€¦ Raychaudhuri, S. (2020). A high-resolution HLA reference panel capturing global population diversity enables multi-ethnic fine-mapping in HIV host response. https://doi.org/10.1101/2020.07.16.20155606

<h2>Four-digit Multi-ethnic HLA v2 (2022)</h2>

<table border="1" cellpadding="5" cellspacing="0">
    <tr>
        <th></th>
        <th></th>
    </tr>
    <tr>
        <td>Number of Samples</td>
        <td>20,349</td>
    </tr>
    <tr>
        <td>Sites (chr6)</td>
        <td>22,733 (570 HLA alleles; 3,449 HLA amino acids; 4,023 SNPs within HLA; 14,691 scaffold SNPs)</td>
    </tr>
    <tr>
        <td>Chromosomes</td>
        <td>6:27,970,031-33,965,553</td>
    </tr>
    <tr>
        <td>Website</td>
        <td>
            <a href="https://github.com/immunogenomics/HLA-TAPAS" style="text-decoration: none">https://github.com/immunogenomics/HLA-TAPAS</a>
        </td>
    </tr>
</table>

<h4>Multi-ethnic HLA v2 Details</h4>

<h5>Samples:</h5>
<ul>
    <li>We added Japanese samples with HLA alleles called with deep target sequencing (n = 723; Hirata et al. Nat Genet. 2019)</li>
    <li>Add Estonian biobank samples with IRB approval (n = 2,233; Mitt et al. Eur J Hum Genet. 2017).</li>
</ul>

<h5>HLA alleles:</h5>
<ul>
    <li>We used gold-standard HLA allele calls (instead of inferred calls from whole genome sequencing) when possible</li>
    <li>We updated and recalled HLA alleles using HLA-LA (Dilthey et al. Bioinformatics. 2019) in Jackson Heart Study (n = 3,026).</li>
    <li>We updated and recalled HLA alleles using HLA-LA in Estonian Biobank (n = 2,233).</li>
</ul>

<h5>Scaffold variants:</h5>
<ul>
    <li>We added population-specific SNP variants which are common in one population but rare in the other populations when they are in 1KG and pass QC, by cross-cohort SNP imputation within the reference panel.</li>
    <li>We removed all SNP variants that are not included in the commonly-used genotyping arrays.</li>
</ul>

<h5>Minimac4 options:</h5>
<ul>
    <li>We optimized --probThreshold, --diffThreshold, --topThreshold parameters in minimac4. Optimized parameters are now used for all HLA panels.</li>
</ul>

<h5>Imputation results:</h5>
<ul>
    <li>We output 2-digit and 4-digit-resolution HLA alleles. We do not output higher resolution HLA alleles in this version.</li>
</ul>

</html>